src/imitation/algorithms/bc.py

```diff
@@ -4,181 +4,176 @@
 action) pairs generated by some expert demonstrator.
 """
 
-import contextlib
+import itertools
+from dataclasses import dataclass
 from typing import Any, Callable, Iterable, Mapping, Optional, Tuple, Type, Union
 
 import gym
 import numpy as np
 import torch as th
-import tqdm.autonotebook as tqdm
-from stable_baselines3.common import policies, utils, vec_env
-
-from imitation.algorithms import base as algo_base
-from imitation.algorithms.base import TransitionMapping
+from stable_baselines3.common import utils, vec_env
+from stable_baselines3.common.policies import ActorCriticPolicy
+
+from imitation.algorithms.base import (
```

Comment:
Per https://github.com/HumanCompatibleAI/imitation/blob/a5b311ff32ea14859eb0f64fe3deef378a73c543/CONTRIBUTING.md we're following the Google Python style guide except where it conflicts with PEP8. In particular, https://google.github.io/styleguide/pyguide.html#22-imports "Use import statements for packages and modules only, not for individual classes or functions."

I don't feel that strongly about this (although I do prefer that style as it makes it more explicit where things are coming from), but if we do change it we should probably do that code-base wide in a separate PR.

Reply:
Ah this rule is new to me. I don't think its bad. I fixed it for now. I think the rule forces us to really think about our module names. E.g. I think we should reconsider the (repeated) usage of `base`, `common` etc. and maybe stick to more specific names.

Reply:
I agree we should try to improve the module names.